### ISSUE TITLE
Performance of call/ec in Racket CS is generally worse than call/cc

### ISSUE THREAD

**Issue by sorawee** [racket/racket#3373]
**What version of Racket are you using?**
7.8 [cs]

**What program did you run?**

```
#lang racket/base
(collect-garbage)
(time
 (for ([_ (in-range 10000000)])
   (call/cc
    (λ (k)
      (+ 1 (k 0))))))
;cpu time: 487 real time: 487 gc time: 7
(collect-garbage)
(time
 (for ([_ (in-range 10000000)])
   (call/ec
    (λ (k)
      (+ 1 (k 0))))))
;cpu time: 794 real time: 794 gc time: 13
```

The program is authored by @yjqww6

**What should have happened?**

`call/ec` should be faster than `call/cc`, or the documentation should be modified to not suggest it as a cheap replacement of `call/cc` to improve performance.

**If you got an error message, please include it here.**

N/A

**Please include any other relevant details**

N/A

**Comment by gus-massa** [comment]
My guess is that this is difficult to fix, but it would be really nice that always `call/ec` is faster than `call/cc`. Is it possible to add this to a wish list?

**Comment by mflatt** [comment]
I'm not sure there's any fundamental reason that `call/ec` should be faster than `call/cc`. Although `call/cc` has to support additional operations, the reverse is also true: `call/ec` has to establish an identity that can be checked relative to the current continuation, and it has to evaluate its body in non-tail position.

So, it depends on the representation of continuations. In BC, the representation is such that the extra functionality of `call/cc` takes much more work. Not so with CS, and the CS representation is all around better: `call/ec` is 2-3 times as fast in CS as in BC. The current implementation of escape continuations in CS does require a little more work to set up the identity, and non-takl evaluation contributes a little (e.g., adding `values` around the `(+ 1 (k 0))` in the `call/cc` case slows it down by more than 10%).

To put it another way: `call/ec` in BC costs the same as 64 `cons`es, while it costs 27 `cons`es in CS (and CS `cons`es are already faster). No one should worry that using `call/ec` will make their program slow.

Of course, it's always possible to make any of the operations faster, and I'm happy for someone to do that. :)

**Comment by yjqww6** [comment]
Another interesting fact is I observed 5%-10% improvements by replacing `(k 0)` with `(call-in-continuation k (λ () 0))` in both call/cc and call/ec cases.(CS)

**Comment by gus-massa** [comment]
I always imagined the `escape` part as a cheap trick to make it faster, not as a security feature to ensure that the continuation is not stored and used later in some code like:

    (let/ec exit
        (f exit)
        (display 'something))
